libbeat/processors/add_docker_metadata/watcher.go

```diff
@@ -164,8 +218,42 @@ func (w *watcher) watch() {
 
 			case <-w.ctx.Done():
 				logp.Debug("docker", "Watcher stopped")
+				w.stopped.Done()
 				return
 			}
 		}
 	}
 }
+
+// Clean up deleted containers after they are not used anymore
+func (w *watcher) cleanupWorker() {
+	for {
+		// Wait a full period
+		time.Sleep(w.cleanupTimeout)
+
+		select {
+		case <-w.ctx.Done():
+			w.stopped.Done()
+			return
+		default:
+			// Check entries for timeout
+			var toDelete []string
+			timeout := time.Now().Add(-w.cleanupTimeout)
+			w.RLock()
+			for key, lastSeen := range w.deleted {
```

Comment:
should we add some debug logging to this? Otherwise very hard to figure out what is still in memory and what was removed.